engine/compiler/clone.go

```diff
@@ -38,6 +38,10 @@ func cloneParams(src manifest.Clone) map[string]string {
 		dst["GIT_SSL_NO_VERIFY"] = "true"
 		dst["PLUGIN_SKIP_VERIFY"] = "true"
 	}
+
+	if src.Retries > 0 {
+		dst["PLUGIN_RETRIES"] = strconv.Itoa(src.Retries)
+	}
```

Comment:
Should we provide a default Retries number?

If I understand correctly, users must set below in their pipeline for this to take effect?
```yaml
clone:
  retries: 3
```